From 43233a7b277e6bd177de73ad36fb89b9d4575b56 Mon Sep 17 00:00:00 2001
From: George Sandu <georgex.sandu@intel.com>
Date: Thu, 14 May 2015 08:29:03 +0200
Subject: [PATCH] Change the bind method for M.2 wwan_noarp driver

For the M.2 modems (enumerated with PID/VID 0x1519, 0x0443)
the IPV6 it's not functional because a MAC address is
defined for this Ethernet interface during the CDC_NCM
driver initialization. The IPV6 stack will use this MAC
address link/local instead of the link/local coming from
the network, resulting in a wrong IPV6 address.
This issue is fixed for the solder down modems (enumerated
with PID/VID 1519, 0452) by using a specific bind method
cdc_ncm_remote_mac_bind. In this method the MAC address
length is set to 1.
This patch is aligning the bind method for the M.2 modems
with the bind method for solder down modems: from
cdc_ncm_bind to cdc_ncm_remote_mac_bind.

Change-Id: If7c97d41638f6734a75b1e27992d55150ef215ae
Signed-off-by: George Sandu <georgex.sandu@intel.com>
---
 drivers/net/usb/cdc_ncm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/usb/cdc_ncm.c b/drivers/net/usb/cdc_ncm.c
index fc486167d0c2..dc0b71a0e893 100644
--- a/drivers/net/usb/cdc_ncm.c
+++ b/drivers/net/usb/cdc_ncm.c
@@ -1259,7 +1259,7 @@ static const struct driver_info wwan_noarp_info = {
 	.description = "Mobile Broadband Network Device (NO ARP)",
 	.flags = FLAG_POINTTOPOINT | FLAG_NO_SETINT | FLAG_MULTI_PACKET
 			| FLAG_WWAN | FLAG_NOARP,
-	.bind = cdc_ncm_bind,
+	.bind = cdc_ncm_remote_mac_bind,
 	.unbind = cdc_ncm_unbind,
 	.check_connect = cdc_ncm_check_connect,
 	.manage_power = usbnet_manage_power,
-- 
1.9.1

